android/src/main/java/com/urbanairship/reactnative/UrbanAirshipReactModule.java

```diff
@@ -127,8 +127,20 @@ public String getName() {
     @ReactMethod
     public void addAndroidListener(String eventName) {
         Logger.info("UrbanAirshipReactModule - Event listener added: " + eventName);
-        listenerCount++;
-        EventEmitter.shared().setEnabled(getReactApplicationContext(), true);
+        EventEmitter.shared().listenerCount++;
+        EventEmitter.shared().addKnownListener(eventName);
```

Comment:
I think you need to add the eventName to the listeners after you flush the pending events. As it is, if an event arrives before you are done flushing, that event might be passed to the listener out of order.